reco_utils/nni/nni_utils.py

```diff
@@ -0,0 +1,58 @@
+import json
+import numpy as np
+import os
+import requests
+import time
+
+NNI_REST_ENDPOINT = 'http://localhost:8080/api/v1/nni'
+NNI_STATUS_URL = NNI_REST_ENDPOINT + '/check-status'
+NNI_TRIAL_JOBS_URL = NNI_REST_ENDPOINT + '/trial-jobs'
+
+
+def get_experiment_status(status_url):
+    nni_status = requests.get(status_url).json()
+    return nni_status['status']
+
+
+def check_experiment_status():
+    while True:
+        time.sleep(20)
+        status = get_experiment_status(NNI_STATUS_URL)
+        if status in ['DONE', 'TUNER_NO_MORE_TRIAL']:
+            break
+        elif status not in ['RUNNING', 'NO_MORE_TRIAL']:
+            raise RuntimeError("NNI experiment failed to complete with status {}".format(status))
+
+
+def check_stopped():
+    while True:
+        time.sleep(20)
+        try:
+            get_experiment_status(NNI_STATUS_URL)
+        except:
+            break
```

Comment:
Erm, if you get an error in the above call your try except will simply swallow it and never pass it on to the user. This will make it super hard to debug.

Reply:
It waits until the NNI experiment completes. NNI status will eventually become 'DONE' or 'TUNER_NO_MORE_TRIAL', or NNI will raise an error.

Reply:
but what happens when status is 'running'?

Reply:
Someone needs to stop (i.e. kill) the experiment; while status is 'running' this function will wait.

Reply:
sorry, i guess i meant status = 'DONE', get_experiment_status will generate an error in that case? do you mean check_experiment_status?

Reply:
If `get_experiment_status()` returns any status it means that the experiment is not "stopped" (in the sense that the server process and the REST interface are active). This is what `check_stopped()` checks. The experiment may have completed (status == 'DONE') but the `nnictl` command still returns info about the experiment and the URL is accessible. 
The purpose of `check_stopped()` is to be called right after `nnictl stop` in order to verify that this command worked. You need to ensure this before you create a new tuning experiment.

Reply:
ah, ok i guess the context for how this was to be used was what I was missing. alright, maybe just a note in the docstring that you must call this after nnictl stop is issued otherwise it will hang.